File: examples/csharp/stripPackage.cs
/**
 * @file    stripPackage.cs
 * @brief   Strips the given package from the given SBML file.
 * @author  Frank T. Bergmann
 * 
 * This file is part of libSBML.  Please visit http://sbml.org for more
 * information about SBML, and the latest version of libSBML.
 */

namespace LibSBMLCSExample
{
  using System;
  using System.IO;
  using libsbmlcs;

  public class stripPackage
  {
    public static void Main (string[] args)
    {
      if (args.Length != 3)
      {
        string myname = Path.GetFileName(Environment.GetCommandLineArgs()[0]);
        Console.WriteLine("Usage: {0} input-filename package-to-strip output-filename", myname);
        Environment.Exit(1);
      }

      string inputFile      = args[0];
      string packageToStrip = args[1];
      string outputFile     = args[2];

      if ( ! File.Exists(inputFile) )
      {
        Console.WriteLine("[Error] {0} : No such file.", inputFile);
        Environment.Exit(1);        
      }

      SBMLReader   reader  = new SBMLReader();
      SBMLWriter   writer  = new SBMLWriter();
      SBMLDocument sbmlDoc = reader.readSBML(inputFile);

      if ( sbmlDoc.getErrorLog().getNumFailsWithSeverity(libsbml.LIBSBML_SEV_ERROR) > 0)
      {
        sbmlDoc.printErrors(); 
        Console.WriteLine("[Error] Cannot read {0}", inputFile);
        Environment.Exit(1);        
      }

	  /* create a new conversion properties structure */
      ConversionProperties props = new ConversionProperties();
	  
	  /* add an option that we want to strip a given package */
	  props.addOption("stripPackage", true, "Strip SBML Level 3 package constructs from the model");
	  
	  /* add an option with the package we want to remove */
	  props.addOption("package", packageToStrip, "Name of the SBML Level 3 package to be stripped");
	  
	  /* perform the conversion */
	  if (sbmlDoc.convert(props) != libsbml.LIBSBML_OPERATION_SUCCESS)
	  {
	  	Console.WriteLine ("conversion failed ... ");
	  	Environment.Exit(3);   
	  }
	  
      writer.writeSBML(sbmlDoc, outputFile);

      Console.WriteLine("[OK] Stripped package '{0}' from {1} and wrote to {2}", packageToStrip, inputFile, outputFile);
    }
  }

}

File: examples/perl/stripPackage.pl
#!/usr/bin/env perl
# -*-Perl-*-
##
## @file    stripPackage.pl
## @brief   Strips the given package from the given SBML file.
## @author  Frank T. Bergmann
## 
##
## This file is part of libSBML.  Please visit http://sbml.org for more
## information about SBML, and the latest version of libSBML.

use File::Basename;
use LibSBML;
no strict;

if ($#ARGV != 2) {
   print  "usage: stripPackage.pl input-filename package-to-strip output-filename\n";
   exit 1;
}

$infile  = $ARGV[0];
$package = $ARGV[1];
$outfile = $ARGV[2];

unless (-e $infile) {
  print("[Error] ", $infile, ": No such file.", "\n");
  exit 1;
}
$reader  = new LibSBML::SBMLReader();
$writer  = new LibSBML::SBMLWriter();
$sbmldoc = $reader->readSBML($infile);

if ($sbmldoc->getNumErrors() > 0) {
  if ($sbmldoc->getError(0)->getErrorId() == $LibSBML::XMLFileUnreadable) {
    # Handle case of unreadable file here.
    $sbmldoc->printErrors();
  }
  elsif ($sbmldoc->getError(0)->getErrorId() == $LibSBML::XMLFileOperationError) {
    # Handle case of other file error here.
    $sbmldoc->printErrors();
  }
  else {
    # Handle other error cases here.
    $sbmldoc->printErrors();
  }
  exit 1;
}
$props = new LibSBML::ConversionProperties();
$props->addOption("stripPackage", true, "Strip SBML Level 3 package constructs from the model");
$props->addOption("package", $package,"Name of the SBML Level 3 package to be stripped");
if ($sbmldoc->convert($props) != $LibSBML::LIBSBML_OPERATION_SUCCESS)
{
  print("[Error] Conversion failed.", "\n");
  exit 3;
}
$writer->writeSBML($sbmldoc, $outfile);
print("[OK] striped package '", $package,"' from ", $infile, " to ", $outfile, "\n");

